mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -197,63 +168,23 @@ impl PutObjectRequest for S3PutObjectRequest {
         let elapsed = self.start_time.elapsed();
         emit_throughput_metric(self.total_bytes, elapsed, "put_object");
 
-        check_response_headers(
-            self.response_headers
-                .lock()
-                .expect("must be able to acquire headers lock")
+        let response_headers_lock = self
+            .response_headers
+            .lock()
+            .expect("must be able to acquire headers lock");
+        let response_headers = response_headers_lock
+            .as_ref()
+            .expect("PUT response headers must be available at this point");
```

Comment:
nit: you can just take the value out here, like this:
```suggestion
        let response_headers = self
            .response_headers
            .lock()
            .expect("must be able to acquire headers lock")
            .take()
            .expect("PUT response headers must be available at this point");
```

Reply:
looks more neat! we consume headers though and making them unavailable for further use from `self.response_headers`, but since it is in `review_and_complete` I believe there is no reason to expect we will ever need them after this point